lib/rules/max-empty-lines/index.js

```diff
@@ -165,25 +165,21 @@ function rule(max, options, context) {
 
 			// TODO: Issue #4985
 			// eslint-disable-next-line no-shadow
-			let result;
-
-			if (/(\r\n)+/g.test(str)) {
-				result = str.replace(/(\r\n)+/g, ($1) => {
-					if ($1.length / 2 > repeatTimes) {
-						return emptyCRLFLines;
-					}
-
-					return $1;
-				});
-			} else {
-				result = str.replace(/(\n)+/g, ($1) => {
-					if ($1.length > repeatTimes) {
-						return emptyLFLines;
-					}
-
-					return $1;
-				});
-			}
+			const result = /(\r\n)+/g.test(str)
+				? str.replace(/(\r\n)+/g, ($1) => {
+						if ($1.length / 2 > repeatTimes) {
+							return emptyCRLFLines;
+						}
+
+						return $1;
+				  })
+				: str.replace(/(\n)+/g, ($1) => {
+						if ($1.length > repeatTimes) {
+							return emptyLFLines;
+						}
+
+						return $1;
+				  });
 
 			return result;
```

Comment:
[suggestion] Does it make sense to inline the `result` variable? We can also remove the `TODO: Issue #4985` comment.

```suggestion
			return /(\r\n)+/g.test(str)
				? str.replace(/(\r\n)+/g, ($1) => {
						if ($1.length / 2 > repeatTimes) {
							return emptyCRLFLines;
						}

						return $1;
				  })
				: str.replace(/(\n)+/g, ($1) => {
						if ($1.length > repeatTimes) {
							return emptyLFLines;
						}

						return $1;
				  });
```

Reply:
Yup, there are 2 places we could do this now, but I wasn't sure if everyone liked the ternary change so I didn't remove the vars :) I can remove them later.

Reply:
Actually @ybiquitous feel free to push any patches to this branch :)

* https://github.com/stylelint/stylelint/pull/5048/files#diff-b3326c81bc17dd9317973d495767164226ba4f78683c2c63a8d5039cd5cf33c3R45
* https://github.com/stylelint/stylelint/pull/5048/files#diff-16a9f35cd6251529efb2d8fa37e0b7a218153be95052f697e35d5cbb6cbcd6edR168

Reply:
Thanks. Maybe, the 1st change may need `/** @type {PostcssResult} */` still.